### ISSUE TITLE
How to align / center a box within the terminal window?

### ISSUE THREAD

**Issue by Davidy22** [ascii-boxes/boxes#86]
[Figlet](http://www.figlet.org/figlet-man.html) has the -lrcx options for aligning the output and -tw for either getting the terminal width or setting an outer width bound for purposes of justfying the final output. -t and -w are meaningless in the context of this program, but it'd be nice if boxes could have some equivalent of figlet's -lrcx flags so that I can have a pretty box with text print in the middle or right of my terminal instead of always on the left. Probably need to do something equivalent to figlet's -t to get the terminal width so that these flags can work.

**Comment by tsjensen** [comment]
Well, *boxes* has `-a` ([docs](https://boxes.thomasjensen.com/boxes-man-1.html#OPTIONS)) which should be very similar to Figlet's -lrcx.  
You can even center Figlet text, e.g. `figlet foo | boxes -a hc -s 50`.

Terminal width is normally something that *boxes* is agnostic of, because *boxes* will always try to match the input text. Additional padding can be specified with `-p`.

Let me know if maybe that already solves your case!

**Comment by tsjensen** [comment]
I just found that on my terminal (Debian / bash), I can

    figlet "Hello World !" | boxes -d peek -a hc -s $COLUMNS

To get a box which spans exactly the terminal width. Don't know who keeps the `$COLUMNS` up-to-date, but it works out of the box.

**Comment by Davidy22** [comment]
The -p and -a flags let me shift the text around inside an oversized box, but I want to realign the box itself to be centered in the terminal, but still fitted to the width of the text. $COLUMNS gets us a bit of the way there, but currently we only have the option to have our box take up the full width, can't center or right justify a smaller box within the width of $COLUMNS.

**Comment by tsjensen** [comment]
Ah, I think I get your point now. Thanks for clarifying.  
Try this.

Centered box with figlet text:

    figlet centered | boxes -d unicornthink | boxes -s $COLUMNS -c x -a hc -i none | sed -e 's/^x/ /'

Right-aligned box with figlet text:

    figlet right-aligned | boxes -d unicornthink | boxes -s $COLUMNS -c x -a hr -i none | sed -e 's/^x/ /'

The basic idea is to call *boxes* twice, the second time with only a few `x` in the first column of the result, which we replace by a space with the `sed` command. In this way, *boxes* centers the box.  
Some googling gave me that one might use `$(tput cols)` instead of `$COLUMNS` for slightly better coverage of different shells.

**Comment by Davidy22** [comment]
well i guess that works, a lotta extra work over just a flag. Actually if this is going to be the recommended approach, maybe there could be a blank "box" that prints nothing, is only for alignment purposes and is appropriately named, so that the command chain gets cut down to:

`figlet right-aligned | boxes -d unicornthink | boxes -s $COLUMNS -d align -a hr

Saving a sed command, making this a lot more discoverable and easier to figure out independently.

**Comment by tsjensen** [comment]
In order to make this easier to discover, I added a [section to the FAQ](https://boxes.thomasjensen.com/faq.html#q7).

While trying out the commands above, I found that a blank box actually causes an error `boxes may not consist entirely of whitespace`. I'll open a new issue as a reminder to myself to find out why that is. Maybe we can enable a new use case for *boxes*.

In general, unix philosophy says that tools should [do one thing, and do it well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well). So, aligning stuff in a terminal window should be a task for another tool. (On Mac, `fmt` is supposed to be able to do that.) It would be hard for *boxes* to reliably determine the terminal width on all of the platforms it runs on. For example, I wouldn't know how to do this on Windows.
